File: src/main/java/com/testsmirk/UserRepository.java
package com.testsmirk;

import org.bson.types.ObjectId;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * Created by Smirk on 2017/3/4.
 */
public interface UserRepository extends MongoRepository<User,Long>{
    User findByUsername(String name);


    User findAllExceptId(String id);

    User findByUserid(String name);
}

File: src/main/java/com/testsmirk/UserRepository.kt
package com.testsmirk

import org.bson.types.ObjectId
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query
import org.springframework.stereotype.Repository

/**
 * Created by Smirk on 2017/3/4.
 */
interface UserRepository : MongoRepository<User, Long> {
    fun findByUsername(name: String): User


}
